src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpSyncNamespaceCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,217 @@
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.SyncNamespace
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.SyncNamespace), Shared]
+    internal sealed class CSharpSyncNamespaceCodeRefactoringProvider :
+        AbstractSyncNamespaceCodeRefactoringProvider<CSharpSyncNamespaceCodeRefactoringProvider, NamespaceDeclarationSyntax, CompilationUnitSyntax>
+    {
+        protected override ImmutableArray<ISymbol> GetDeclaredSymbolsInContainer(SemanticModel semanticModel, SyntaxNode node, CancellationToken cancellationToken)
+        {
+            var declarations = GetMemberDeclarationsInContainer(node);
+            var builder = ArrayBuilder<ISymbol>.GetInstance();
+            foreach (var declaration in declarations)
+            {
+                var symbol = semanticModel.GetDeclaredSymbol(declaration, cancellationToken);
+                builder.AddIfNotNull(symbol);
+            }
+            return builder.ToImmutableAndFree();
+        }
+
+        private IReadOnlyList<MemberDeclarationSyntax> GetMemberDeclarationsInContainer(SyntaxNode node)
+        {
+            if (node is NamespaceDeclarationSyntax namespaceDecl)
+            {
+                return namespaceDecl.Members;
+            }
+            else if (node is CompilationUnitSyntax compilationUnit)
+            {
+                return compilationUnit.Members;
+            }
+            else
+            {
+                return default;
+            }
+        }
+
+        private static NameSyntax GetQualifiedNameSyntax(NameSyntax node) 
+            => node.Parent is QualifiedNameSyntax qualifiedName && qualifiedName.Right == node ? qualifiedName : node;
+
+        protected override bool TryGetReplacementSyntax(SyntaxNode reference, ImmutableArray<string> namespaceParts, out SyntaxNode old, out SyntaxNode @new)
+        {
+            if (reference is SimpleNameSyntax nameRef)
+            {
+                var outerMostNode = GetQualifiedNameSyntax(nameRef);
+                old = outerMostNode;
+
+                // If no namespace is specified, we just find and return the full NameSyntax.
+                if (outerMostNode == nameRef || namespaceParts.IsDefault)
+                {                        
+                    @new = outerMostNode;  
+                }
+                else
+                {
+                    var aliasQualifier = GetAliasQualifierOpt(outerMostNode);
+                    var qualifiedNamespaceName = CreateNameSyntax(namespaceParts, aliasQualifier, namespaceParts.Length - 1);
+
+                    // We might lose some trivia associated with nodes within `outerMostNode`.  
+                    @new = SyntaxFactory.QualifiedName(qualifiedNamespaceName, nameRef).WithTriviaFrom(outerMostNode);
+                } 
+                return true;
+            }
+            else
+            {
+                old = @new = null;
+                return false;
```

Comment:
when does this happen?

Reply:
This is just to make the code robust in case I misunderstood the code. I looked at the `ReferenceFinder` implementation, seems the search is based on matching token text, so my guess is we will always get a `SimpleNameSyntax` with returned span, but I'm not 100% certain.